overlay/github.nix

```diff
@@ -0,0 +1,45 @@
+{ gh, git, writeShellApplication }:
+{
+  # autorelease <release-assets> <release-notes> [release-tag]
+  #
+  # release-assets
+  #   A path to the directory with release assets, typically something like "$(nix build .#release)"
+  # release-notes
+  #   Either a path to the file or bare text to use as release notes.
+  # release-tag (optional)
+  #   A tag for the release to be pushed and also to be used as a release title.
+  #   Default value to be used is 'auto-release'.
+  #
+  # If 'release-tag' is 'autorelease' or 'PRERELEASE' env variable is set to 'true'
+  # created release is marked as 'prerelease'.
+  #
+  # Usage examples:
+  # autorelease "$(nix build .#release)" "Automatic release on "$(date +\"%Y%m%d%H%M\")""
+  #
+  # autorelease "$(nix build .#release)" ./release-notes.md "v1.0"
+  autorelease = writeShellApplication {
+    name = "autorelease";
+    runtimeInputs = [ gh git ];
+    text = ''
+      release_assets="$1"
+      release_notes="$2";
+      release_tag="''${3:-auto-release}"
+
+      # gh expects 'GH_TOKEN' env variable to be set
+      # Delete release if it exists
+      gh release delete "$release_tag" --yes || true
```

Comment:
First of all, I'm not familiar with this gh cli, so please correct me if I'm wrong. 
For me this delete seems too destructive. If the problem is really the presence of an existing release with the same name, I would opt to be deleted by a human being.

Reply:
Done.
I didn't want to complicate the arguments parser since it's quite painful to do non-trivial arg parsing in bash, so I decided to do this via dedicated env variable